### PR TITLE
system - added recognition for Jetson Nano and Tegra X1

### PR DESCRIPTION
**Pull request by joolswills**
* by user request - not officially supported

### CODE REVIEW

File: scriptmodules/system.sh [hunk 1]
```diff
@@ -377,11 +377,26 @@ function get_platform() {
...
+                # jetson nano and tegra x1 can be identified via /sys/firmware/devicetree/base/model
+                local model_path="/sys/firmware/devicetree/base/model"
+                if [[ -f "$model_path" ]]; then
+                    # ignore end null to avoid bash warning
+                    local model=$(tr -d '\0' <$model_path)
+                    case "$model" in
+                        "NVIDIA Jetson Nano Developer Kit")
+                            __platform="jetson-nano"
+                            ;;
+                        *icosa*)
```
Comment (joolswills) [inline scriptmodules/system.sh:389]: Wasn't sure if this was requiring a partial match or not.

Comment (theofficialgman) [reply scriptmodules/system.sh:389]: it should be icosa and that's it

**Comment by joolswills** [comment]
This is hardcoded for only aarch64 for these boards btw. I don't have anything to test with. If there's more arm x11 + GL boards I may rework it somewhat. I have plans to split out the systems configs after more refactoring anyway.

File: scriptmodules/system.sh (hunk 1 above)
Comment (joolswills) [reply scriptmodules/system.sh:389]: I'll remove the wildcards

Comment (joolswills) [reply scriptmodules/system.sh]: Done.

Comment (joolswills) [reply scriptmodules/system.sh]: Thanks.

**Comment by theofficialgman** [comment]
nvidia renames the model if its not a tegra x1 so there shouldn't be any work of it not being aarch64

**Comment by joolswills** [comment]
Thanks. Please let me know if this works and I'll merge it.

**Comment by theofficialgman** [comment]
Just finished testing your latest changes (37c14a2) and built a system that it compatible (lr-parallel-n64). Seems to work as expected.

**Comment by joolswills** [comment]
I just made one further change. Since they both share the same platform specs as far as we are concerned, I made the jetson nano share the tegra-x1 platform config. Just a bit of deduplication

**Comment by joolswills** [comment]
Thanks. My last change shouldn't affect anything so I'll merge in a bit.

**Comment by theofficialgman** [comment]
that's fine. jetson nano really only differs in number of gpu cuda cores which isn't affected here

**Comment by joolswills** [comment]
Interesting. I know very little about these systems. Maybe I should get a second hand switch to have a play.

**Comment by theofficialgman** [comment]
you have to get an original switch model (not the new switch) since you can't hack those

**Pull request merged**